haec-user-service/src/modules/user/services/user.service.ts

```diff
@@ -46,14 +46,51 @@ export class UserService {
           email: input.email,
         },
       });
-
       if (await bcrypt.compare(input.password, user.hashedPass)) {
         return true;
       }
     } catch (error) {
       console.error(error);
     }
-
     return false;
   }
+
+// This will log the user ** IN **
+
+  public async loginUser(input: {
+    email: string;
+    password: string;
+  }): Promise<string> {
+    const verified = await this.verifyUser({ email: input.email, password: input.password})
+    if (verified == true){
+      // Generates random 21 character token with Crypto
+      const Crypto = require('crypto')
+      function randomString(size = 21) {
+        return Crypto
+          .randomBytes(size)
+          .toString('base64')
+          .slice(0, size)
+      }
+      const token = randomString()
+      await this.prisma.user.update({
+        where: { email: input.email },
+        data: { token: token } })
+      return token
+  }
+  else {
+    console.log("error")
+    throw new Error()
+  }
+  }
+
+// This will log the user ** OUT **
+
+public async logoutUser(input: {
+  email: string;
+}): Promise<string> {
+  await this.prisma.user.update({
+    where: { email: input.email },
+    data: { token: null } })
+  return
```

Comment:
I think this syntax doesn't actually make sense. When you do `return` without anything after it, this is the equivalent of writing `return undefined` (which is fine btw, nothing wrong with a function returning `undefined`).

In your case here, this'll get you in trouble because your defined return is `Promise<string>` not `Promise<void>`. There's a slight nuance here with Typescript when you're returning "nothing" -- which is why you have to return `Promise<void>` and not `Promise<undefined>`.

Reply:
Also -- you're not required to have a `return` statement. They're optional (especially if your function is a `Promise<void>`).

Reply:
That makes sense. I'm not returning data, I'm just logging them out.